aosp_diff/preliminary/external/drm_hwcomposer/0005-Fix-drm-hwc-import-buffer-failure-issue.patch

```diff
@@ -0,0 +1,94 @@
+From c1846aa23e1a48a27ffaa2813ccd39d527b565dc Mon Sep 17 00:00:00 2001
+From: "Li, HaihongX" <haihongx.li@intel.com>
+Date: Tue, 19 Oct 2021 09:24:48 +0800
+Subject: [PATCH] Fix drm-hwc import buffer failure issue
+
+Set format is DRM_FORMAT_NV12 if DRM_FORMAT_NV12_Y_TILED_INTEL.
+Don't close the same handle several times.
+
+Tracked-On: OAM-99497
+Signed-off-by: Li, HaihongX <haihongx.li@intel.com>
+---
+ drm/DrmFbImporter.cpp   | 15 ++++++++++++++-
+ drm/DrmPlane.cpp        |  2 +-
+ include/drmhwcomposer.h |  1 +
+ 3 files changed, 16 insertions(+), 2 deletions(-)
+
+diff --git a/drm/DrmFbImporter.cpp b/drm/DrmFbImporter.cpp
+index 592fc48..e6e7bf9 100644
+--- a/drm/DrmFbImporter.cpp
++++ b/drm/DrmFbImporter.cpp
+@@ -40,6 +40,7 @@ auto DrmFbIdHandle::CreateInstance(hwc_drm_bo_t *bo, GemHandle first_gem_handle,
+   local->gem_handles_[0] = first_gem_handle;
+   int32_t err = 0;
+ 
++  uint32_t num_planes = 0;
+   /* Framebuffer object creation require gem handle for every used plane */
+   for (int i = 1; i < local->gem_handles_.size(); i++) {
+     if (bo->prime_fds[i] > 0) {
+@@ -52,6 +53,7 @@ auto DrmFbIdHandle::CreateInstance(hwc_drm_bo_t *bo, GemHandle first_gem_handle,
+         }
+       } else {
+         local->gem_handles_.at(i) = local->gem_handles_[0];
++        num_planes++;
+       }
+     }
+   }
+@@ -72,6 +74,15 @@ auto DrmFbIdHandle::CreateInstance(hwc_drm_bo_t *bo, GemHandle first_gem_handle,
+                         &local->gem_handles_[0], &bo->pitches[0],
+                         &bo->offsets[0], &local->fb_id_, 0);
+   } else {
++    if (bo->format == DRM_FORMAT_NV12_Y_TILED_INTEL) {
++      for (uint32_t i = num_planes; i < HWC_DRM_BO_MAX_PLANES; i++) {
```

Comment:
these two 'for' loop can reduce to one? the loop conditions are same.

Reply:
Optimize done.